extensions/amp-access/0.1/amp-access.js

```diff
@@ -93,8 +101,20 @@ export class AccessService {
     /** @private @const {!Cid} */
     this.cid_ = cidFor(this.win);
 
+    /** @private @const {!Viewer} */
+    this.viewer_ = viewerFor(this.win);
+
+    /** @private @const {!DocumentState} */
+    this.docState_ = documentStateFor(this.win);
+
+    /** @private @const {!Viewport} */
+    this.viewport_ = viewportFor(this.win);
+
     /** @private {!Promise<string>|undefined} */
     this.readerIdPromise_ = undefined;
+
+    /** @private {!Promise|undefined} */
+    this.reportViewPromise_ = undefined;
```

Comment:
Is there for a reason for preferring `undefined` over `null`? The type hint could be shorter with `{?Promise}`...